pkg/controllers/nodeclass/status/images.go

```diff
@@ -112,6 +113,16 @@ func (r *NodeImageReconciler) Reconcile(ctx context.Context, nodeClass *v1beta1.
 	ctx = log.IntoContext(ctx, log.FromContext(ctx).WithName(nodeImageReconcilerName))
 	logger := log.FromContext(ctx)
 
+	// validate FIPS + useSIG
+	fipsMode := nodeClass.Spec.FIPSMode
+	useSIG := options.FromContext(ctx).UseSIG
+	if lo.FromPtr(fipsMode) == v1beta1.FIPSModeFIPS && !useSIG {
+		nodeClass.Status.Images = nil
+		nodeClass.StatusConditions().SetFalse(v1beta1.ConditionTypeImagesReady, "SIGRequiredForFIPS", "FIPS images require UseSIG to be enabled, but UseSIG is false (note: UseSIG is only supported in AKS managed NAP)")
+		logger.Info("FIPS images require SIG", "error", fmt.Errorf("FIPS images require UseSIG to be enabled, but UseSIG is false"))
```

Comment:
nit: I'd update the error message with the same note as in the condition:
```suggestion
		logger.Info("FIPS images require SIG", "error", fmt.Errorf("FIPS images require UseSIG to be enabled, but UseSIG is false (note: UseSIG is only supported in AKS managed NAP)"))
```